components/script/dom/node.rs

```diff
@@ -422,14 +452,26 @@ impl Node {
     }
 
     /// Return all registered mutation observers for this node.
+    /// XXX(ferjm) This should probably be split into two functions,
+    /// `registered_mutation_observers`, which returns an Option or
+    /// an empty slice or something, and doesn't create the rare data,
+    /// and `registered_mutation_observers_mut`, which does lazily
+    /// initialize the raredata.
     pub fn registered_mutation_observers(&self) -> RefMut<Vec<RegisteredObserver>> {
-        self.mutation_observers.borrow_mut()
+        RefMut::map(self.ensure_rare_data(), |rare_data| {
```

Comment:
Ouch, yeah, I missed the details of this always allocating the rare data even if empty. Please file an issue to fix this.